src/types.rs

```diff
@@ -89,6 +89,19 @@ impl Level {
             }
         }
     }
+
+    pub fn is_sokoban_solved(&self) -> bool {
+        for entity in &self.entities {
+            if let Entity::SokoBox(soko_box) = entity {
+                let _t = &self.map[[soko_box.coords.y, soko_box.coords.x]];
+                if let Tile::Goal = _t {
+                } else {
+                    return false;
+                }
+            }
+        }
+        true
+    }
```

Comment:
I think this could be more rust-y doing something more functional:

```
    pub fn is_sokoban_solved(&self) -> bool {
        self.entities
            .iter()
            .filter(|ent| matches!(ent, Entity::SokoBox(_)))
            .all(|ent| {
                if let Entity::SokoBox(soko_box) = ent {
                    let tile = &self.map[[soko_box.coords.y, soko_box.coords.x]];
                    matches!(tile, Tile::Goal)
                } else {
                    false // This line should never be reached due to the filter
                }
            })
    }
```

Reply:
ok i see the rizz now that i did some more exploring rust docs